src/com/reader/AbstractReader.java

```diff
@@ -0,0 +1,43 @@
+package com.reader;
+
+import com.sudoku.SudokuBoard;
+
+import java.util.Scanner;
+
+/**
+ * Class that contains reader methods to read different types of Files
+ */
+public abstract class AbstractReader {
+    protected String filePath;
+    protected Scanner scanner = null;
+    public static final int ASCCI_ZERO = "0".charAt(0);
+
+    public AbstractReader(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Method to get the content of file
+     *
+     * @return content
+     */
+    public abstract String getContentFile();
+
+    /**
+     * Method to return a SudokuBoard
+     *
+     * @return SudokuBoard
+     */
+    public SudokuBoard getSudokuBoard() {
```

Comment:
you need to document better, put more detail on the description adn the parameters of the methods